std/format/internal/write.d

```diff
@@ -28,15 +28,19 @@ package(std.format):
     `bool`s are formatted as `"true"` or `"false"` with `%s` and as `1` or
     `0` with integral-specific format specs.
  */
-void formatValueImpl(Writer, T, Char)(auto ref Writer w, T obj, scope const ref FormatSpec!Char f)
+void formatValueImpl(Writer, T, Char)(auto ref Writer w, const(T) obj, scope const ref FormatSpec!Char f) // const(T) reduces number of instances
 if (is(BooleanTypeOf!T) && !is(T == enum) && !hasToString!(T, Char))
 {
-    BooleanTypeOf!T val = obj;
+    BooleanTypeOf!T val = obj;  // TODO: remove and pass obj to val directly in line below?
+    formatValueImplBool!(Writer, Char)(w, val, f);
+}
 
+void formatValueImplBool(Writer, Char)(auto ref Writer w, const(bool) val, scope const ref FormatSpec!Char f)
```

Comment:
I fail to understand how factoring code out into a new template declaration reduces template bloat

Reply:
Oh, I understand now. Code that is the same for each template instance should be factored out so that the generated code is smaller.

Reply:
Honestly, I think that such optimizations should be benchmarked on their own. It is hard to assess the impact of this optimization when it is bundled with others. I suggest splitting this PR into different ones and assess the gains on their own. There must be a proven benefit to accept such code surgery.

Reply:
Ok.